<a href="https://colab.research.google.com/github/SinghNavnoor/CodingDojo_Project_2/blob/main/Project_2_propposal.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Dataset Choice 1:

In [1]:
import pandas as pd
df = pd.read_csv('/content/adult.csv')

df.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [4]:
df.shape

(48842, 15)

In [2]:
df.duplicated().sum()

52

In [ ]:
df.isna().sum()

1. Source of data:

- Kaggle

- Link: https://www.kaggle.com/datasets/wenruliu/adult-income-dataset

2. Brief description of data
- Data is about the different aspect of an adult's life. From their gender to educationa level, income and much more. 

3. What is the target?

- The target could be multiple columns. I will be focusing on Race or gender as the Target. The initial target was Income however I would like to do a classification model so depending on which target (Race or gender) give me a more interesting outcome I will go with that for the final target. 

4. What does one row represent? (A person?  A business?  An event? A product?)
- One row represents an individual. 

5. Is this a classification or regression problem?
- Classification

6. How many features does the data have?
- 15



7. How many rows are in the dataset?
- 48,842

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
- It will be challenging to impute null values iin columns like workclass and occupation. I would like to find a Machine Learning solution to impute the missing data, instead of using most_frequent strategy. 
- The Income columns, while representing numerical values, has letters like K to represent thousand and less than, greater than and equal to signs that would need to be dealth with. 
- It will be important to standarize the data b/c there are columns like capital gain and capital loss that either have large numbers or just sttraight zero. 
- I will have to make sure that columns like marital status and occupation have clear classes and then one hot encode them for the model creation. 


#Dataset Choice 2:

In [6]:
df2 = pd.read_csv('/content/wines_SPA.csv')

df2.head()

winery           wine  year  rating  num_reviews country  \
0  Teso La Monja          Tinto  2013     4.9           58  Espana   
1         Artadi  Vina El Pison  2018     4.9           31  Espana   
2   Vega Sicilia          Unico  2009     4.8         1793  Espana   
3   Vega Sicilia          Unico  1999     4.8         1705  Espana   
4   Vega Sicilia          Unico  1996     4.8         1309  Espana   

             region   price                  type  body  acidity  
0              Toro  995.00              Toro Red   5.0      3.0  
1    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0

In [8]:
df2.duplicated().sum()

5452

In [9]:
df2.isna().sum()

winery            0
wine              0
year              2
rating            0
num_reviews       0
country           0
region            0
price             0
type            545
body           1169
acidity        1169
dtype: int64

In [11]:
df2.shape

(7500, 11)

1. Source of data
- Kaggle, https://www.kaggle.com/datasets/fedesoriano/spanish-wine-quality-dataset

2. Brief description of data
- The data looks into various aspects of wine. The exploration goes from how old the wine is, to the region of origin, price, type etc. 

3. What is the target?
- Price

4. What does one row represent? (A person?  A business?  An event? A product?)
- Wine


5. Is this a classification or regression problem?
- regression


6. How many features does the data have?
- 11

7. How many rows are in the dataset?
- 7500


8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
- I see that One-Hot encoding all the categorical variables will increase number of columns the model has to deal with, hence longer time in model building.
- It will be hard to decide whether a column like Winery is important to the model building or not. The reason being that 